students/dave_arasim/session05/mailroom3.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python
+'''Mailroom Part 3 for Session 5, Python 210
+Written by David K. Arasim - 11/05/18'''
+
+# Imported modules
+import platform
+import os
+from collections import defaultdict
+
+# Global variables
+this_os = platform.system()
+quit_option = False
+
+# Form letter text
+form_text = ['Dear {form_name},\n',
+             'Thank you for your generous donation of ${form_amt:.2f}\n',
+             'Please consider this e-mail as record of your donation for tax deduction purposes.\n',
+             'Sincerely, Your Local Charity']
+form_text_str = ''.join(x for x in form_text) # Comprehension
+
+#############################################################################################
+# Function section
+
+def main():
+    global quit_option
+
+    # Build the Donor Dictionary from a text file
+    donor_dict = defaultdict(list)
+    build_donor_dict(donor_dict)
+
+    # Build switch_dict with a dict comprehension
+    switch_options = [1, 2, 3, 4]
+    switch_functions = [send_thank_you, create_report, send_all_thanks, quit_option_true]
+    switch_dict = {option: function for option, function in zip(switch_options, switch_functions)}
+
+    while not(quit_option):
```

Comment:
Minor: `not quit_option`, not `not(quit_option)`, which looks like a function call, but `not` is not a function. When not is applied to a compound boolean expression, parentheses should be used, but after a blank space. E.g., `not (a or b)`.